Bump clap to v4

tranzystorek-io opened the pull request.

dimo414 reviewed (dismissed):
Thanks for the PR!

tranzystorek-io commented on Cargo.toml:
```diff
@@ -25,7 +25,7 @@ debug = ['serde_json']
 [dependencies]
 anyhow = "1.0"
 bincode = "1.3.1"
-clap = { version = "3.0", default_features = false, features = ["std", "derive", "env"] }
```
Corrected

tranzystorek-io commented on src/main.rs:
```diff
@@ -98,56 +98,55 @@ fn run(cli: Cli) -> Result<i32> {
 }
 
-#[derive(Parser)]
-#[clap(setting = AppSettings::DeriveDisplayOrder)]
```
`AppSettings` is deprecated and this is now the default (and seemingly the only option)

dimo414 commented:
Thanks! I'm in the middle of a few changes I'd like to land first, but I'll merge this PR before cutting a 0.6.0 release.

dimo414 commented:
If you'd like to rebase on top of the changes I just pushed I can merge this and cut 0.6.0 :)

tranzystorek-io commented:
Had to adjust slightly:

- added custom value names to make help more readable
- switched `Option<Vec>` args to `Vec`, as it does the same thing
- fixed `--modtime` alias to `--use-file-modtime` (hyphens)

dimo414 commented on src/main.rs:
```diff
@@ -105,61 +105,60 @@ fn run(cli: Cli) -> Result<i32> {
-    // Includes the last modification time of the given file(s) in the cache key,
+    /// Includes the last modification time of the given file(s) in the cache key,
     /// so that the same command run with different modtimes for the given files caches separately
-    #[clap(long, visible_alias = "use_file_modtime", multiple_occurrences=true, use_value_delimiter=false)]
```
Can these settings be preserved?

tranzystorek-io replied:
These are the default now

dimo414 replied:
Well gee, I guess [this line](https://github.com/dimo414/bkt/blob/4b6808d5d43543fa0a45356c5d58f24235abadd2/README.md?plain=1#L112) was never true.... :)

dimo414 commented on src/main.rs:
```diff
@@ -52,12 +52,12 @@ fn run(cli: Cli) -> Result<i32> {
     }
 
-    let envs = cli.env.into_iter().flatten().collect::<Vec<_>>();
+    let envs = cli.env;
```
You can do away with these assignments now, just read the field directly since we're not constructing a separate vec.

tranzystorek-io replied:
Just felt right to keep it in a new variable, but I can change that if you like
